pkg/unidling/controller.go

```diff
@@ -0,0 +1,178 @@
+package unidling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	kextapi "k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kextclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/extensions/unversioned"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/util/unidling"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
+)
+
+type UnidlingController struct {
+	controller          *framework.Controller
+	scaleNamespacer     kextclient.ScalesGetter
+	endpointsNamespacer kclient.EndpointsNamespacer
+}
+
+func NewUnidlingController(scaleNS kextclient.ScalesGetter, endptsNS kclient.EndpointsNamespacer, evtNS kclient.EventNamespacer, resyncPeriod time.Duration) *UnidlingController {
+	fieldSet := fields.Set{}
+	fieldSet["reason"] = unidling.NeedPodsReason
+	fieldSelector := fieldSet.AsSelector()
+
+	unidlingController := &UnidlingController{
+		scaleNamespacer:     scaleNS,
+		endpointsNamespacer: endptsNS,
+	}
+
+	_, controller := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
```

Comment:
I think you can just avoid relist and resync altogether.  If we miss an unidle it's not the end of the world.